gp-conditional-pricing/gpcp-fuzzy-comparison.php

```diff
@@ -0,0 +1,190 @@
+						}
+
+						// Ensure fieldIds are all numbers
+						self.fieldId = self.fieldId.map( function( fieldId ) {
+							if ( typeof fieldId === 'string' ) {
+								fieldId = parseInt( fieldId );
+							}
+							return fieldId;
+						} );
+
+						return self.fieldId.indexOf( fieldId ) !== -1;
+					};
+
+					self.init = function() {
+						gform.addFilter('gform_is_value_match', function (isMatch, formId, rule) {
+							if ( rule.operator !== 'is' ) {
+								return isMatch;
+							}
+							var source = jQuery( '#input_' + formId + '_' + rule.fieldId );
+							if ( source ) {
+								return self.fuzzyMatch( source.val(), rule.value );
+							}
+							return isMatch;
+						});
+					};
+
+					self.normalizeString = function( str ) {
+						return str.normalize("NFD").replace( /[\u0300-\u036f]/g, "" ).toLowerCase();
+					};
+
+					self.levenshtein = function( a, b ) {
+						const an = a.length;
+						const bn = b.length;
+						if (an === 0) return bn;
+						if (bn === 0) return an;
+						const matrix = [];
+						for (let i = 0; i <= bn; i++) {
+							matrix[i] = [i];
+						}
+						for (let j = 0; j <= an; j++) {
+							matrix[0][j] = j;
+						}
+						for (let i = 1; i <= bn; i++) {
+							for (let j = 1; j <= an; j++) {
+								if (b.charAt(i - 1) === a.charAt(j - 1)) {
+									matrix[i][j] = matrix[i - 1][j - 1];
+								} else {
+									matrix[i][j] = Math.min(
+										matrix[i - 1][j - 1] + 1,
+										Math.min(
+											matrix[i][j - 1] + 1,
+											matrix[i - 1][j] + 1
+										)
+									);
+								}
+							}
+						}
+						return matrix[bn][an];
+					};
+
+					self.fuzzyMatch = function( input, target ) {
+						const normalizedInput  = self.normalizeString( input );
+						const normalizedTarget = self.normalizeString( target );
+						const distance         = self.levenshtein( normalizedInput, normalizedTarget );
+						return distance <= self.threshold;
+					};
+
+					self.init();
+				};
+
+			} )( jQuery );
+		</script>
+		<?php
+	}
+
+	public function add_init_script( $form ) {
+		if ( ! $this->is_applicable_form( $form ) ) {
+			return;
+		}
+
+		$args = array(
+			'formId'    => $this->_args['form_id'],
+			'fieldId'   => $this->_args['field_id'],
+			'threshold' => $this->_args['threshold'],
+		);
+
+		$script = 'new GPCP_Fuzzy_Match( ' . json_encode( $args ) . ' );';
+		$slug   = implode( '_', array( strtolower( __CLASS__ ), $this->_args['form_id'], $this->_args['field_id'] ) );
+
+		GFFormDisplay::add_init_script( $form['id'], $slug, GFFormDisplay::ON_PAGE_RENDER, $script );
+	}
+
+	public function is_applicable_form( $form ) {
+		$form_id = isset( $form['id'] ) ? $form['id'] : $form;
+		return empty( $this->_args['form_id'] ) || (int) $form_id === (int) $this->_args['form_id'];
+	}
+}
+
+# Configuration
+new GPCP_Fuzzy_Match( array(
+	'form_id'   => 1,  // Replace with your form ID
+	'field_id'  => 2,  // Replace with your field ID
```

Comment:
Let's clarify that this is ID of the field for which the conditional logic should be fuzzy. I think some customers may think that it might be the ID of the field whose value the conditional logic field is being compared to.